io/src/main/scala/fs2/io/file/Files.scala

```diff
@@ -0,0 +1,510 @@
+    * Reads all data from the file at the specified `java.nio.file.Path`.
+    */
+  def readAll(path: Path, chunkSize: Int): Stream[F, Byte]
+
+  /**
+    * Reads a range of data synchronously from the file at the specified `java.nio.file.Path`.
+    * `start` is inclusive, `end` is exclusive, so when `start` is 0 and `end` is 2,
+    * two bytes are read.
+    */
+  def readRange(path: Path, chunkSize: Int, start: Long, end: Long): Stream[F, Byte]
+
+  /**
+    * Set file permissions from set of [[PosixFilePermission]].
+    *
+    * Note: this will only work for POSIX supporting file systems.
+    */
+  def setPermissions(path: Path, permissions: Set[PosixFilePermission]): F[Path]
+
+  /**
+    * Returns the size of a file (in bytes).
+    */
+  def size(path: Path): F[Long]
+
+  /**
+    * Creates a stream containing the path of a temporary file.
+    *
+    * The temporary file is removed when the stream completes.
+    */
+  def tempFileStream(
+      dir: Path,
+      prefix: String = "",
+      suffix: String = ".tmp",
+      attributes: Seq[FileAttribute[_]] = Seq.empty
+  ): Stream[F, Path]
+
+  /**
+    * Creates a resource containing the path of a temporary file.
+    *
+    * The temporary file is removed during the resource release.
+    */
+  def tempFileResource(
+      dir: Path,
+      prefix: String = "",
+      suffix: String = ".tmp",
+      attributes: Seq[FileAttribute[_]] = Seq.empty
+  ): Resource[F, Path]
+
+  /**
+    * Creates a stream containing the path of a temporary directory.
+    *
+    * The temporary directory is removed when the stream completes.
+    */
+  def tempDirectoryStream(
+      dir: Path,
+      prefix: String = "",
+      attributes: Seq[FileAttribute[_]] = Seq.empty
+  ): Stream[F, Path]
+
+  /**
+    * Creates a resource containing the path of a temporary directory.
+    *
+    * The temporary directory is removed during the resource release.
+    */
+  def tempDirectoryResource(
+      dir: Path,
+      prefix: String = "",
+      attributes: Seq[FileAttribute[_]] = Seq.empty
+  ): Resource[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s contained in a given file tree. Depth is unlimited.
+    */
+  def walk(start: Path): Stream[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s contained in a given file tree, respecting the supplied options. Depth is unlimited.
+    */
+  def walk(start: Path, options: Seq[FileVisitOption]): Stream[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s contained in a given file tree down to a given depth.
+    */
+  def walk(start: Path, maxDepth: Int, options: Seq[FileVisitOption] = Seq.empty): Stream[F, Path]
+
+  /**
+    * Writes all data to the file at the specified `java.nio.file.Path`.
+    *
+    * Adds the WRITE flag to any other `OpenOption` flags specified. By default, also adds the CREATE flag.
+    */
+  def writeAll(
+      path: Path,
+      flags: Seq[StandardOpenOption] = List(StandardOpenOption.CREATE)
+  ): Pipe[F, Byte, INothing]
```

Comment:
I wonder if we could introduce another definition with variance, of the following form: 
```Scala
def writeAll[ InF[x] >: F[x], OutF[x] <: F[x]] (-----): InF[Byte] => OutF[INothing]
```

Reply:
Perhaps, though if doing so, I'd rather just avoid aliases completely and return `Stream[F, Byte] => Stream[F, INothing]`, which has the right variance. This would be a different PR though -- one that deprecates `Pipe` completely.